sql/fulltext/fulltext.go

```diff
@@ -396,6 +396,101 @@ func RebuildTables(ctx *sql.Context, tbl sql.IndexAddressableTable, db Database)
 	return CreateFulltextIndexes(ctx, db, tbl, predeterminedNames, indexDefs...)
 }
 
+// DropColumnFromTables removes the given column from all of the Full-Text indexes, which will trigger a rebuild if the
+// index spans multiple columns, but will trigger a deletion if the index spans that single column. The column name is
+// case-insensitive.
+func DropColumnFromTables(ctx *sql.Context, tbl sql.IndexAddressableTable, db Database, colName string) error {
+	// Check the interfaces on the parameters
+	dropper, ok := db.(sql.TableDropper)
+	if !ok {
+		return sql.ErrIncompleteFullTextIntegration.New()
+	}
+	idxAlterable, ok := tbl.(sql.IndexAlterableTable)
+	if !ok {
+		return sql.ErrIncompleteFullTextIntegration.New()
+	}
+
+	lowercaseColName := strings.ToLower(colName)
+	configTableReuse := make(map[string]bool)
+	predeterminedNames := make(map[string]IndexTableNames)
+	var indexDefs []sql.IndexDef
+
+	// Load the indexes to search for Full-Text indexes
+	indexes, err := tbl.GetIndexes(ctx)
+	if err != nil {
+		return err
+	}
+	for _, index := range indexes {
+		// Skip all non-Full-Text indexes
+		if !index.IsFullText() {
+			continue
+		}
+		// Store the index definition so that we may recreate it below
+		ftIndex := index.(Index)
+		tableNames, err := ftIndex.FullTextTableNames(ctx)
+		if err != nil {
+			return err
+		}
+		predeterminedNames[ftIndex.ID()] = tableNames
+		// Iterate over the columns to search for the given column
+		exprs := ftIndex.Expressions()
+		var indexCols []sql.IndexColumn
+		for _, expr := range exprs {
+			exprColName := strings.TrimPrefix(expr, ftIndex.Table()+".")
+			// Skip this column if it matches our given column
+			if strings.ToLower(exprColName) == lowercaseColName {
+				continue
+			}
+			indexCols = append(indexCols, sql.IndexColumn{
+				Name:   exprColName,
+				Length: 0,
+			})
+		}
+		if len(indexCols) > 0 {
+			// This index will continue to exist, so we want to preserve the config table
+			indexDefs = append(indexDefs, sql.IndexDef{
+				Name:       ftIndex.ID(),
+				Columns:    indexCols,
+				Constraint: sql.IndexConstraint_Fulltext,
+				Storage:    sql.IndexUsing_Default,
+				Comment:    ftIndex.Comment(),
+			})
+			configTableReuse[tableNames.Config] = true
+		} else {
+			// This index will be deleted, so we should delete the config table if no other indexes will reuse the table
+			if _, ok := configTableReuse[tableNames.Config]; !ok {
+				configTableReuse[tableNames.Config] = false
+			}
+		}
```

Comment:
imo i'd just use a list for the configs to keep, map makes it more difficult to iterate below

Reply:
The map is so that we may know which config tables are connected to our indexes. So we want all config tables in the map. If an existing index references a config table, then we set that table to `true` to indicate that the config table is being reused. If an index is being deleted, then it does not reuse the config table, and the config table should be marked for deletion (if it's not already marked for reuse). We then check if the entry already exists, as we don't want to overwrite an entry that is being reused.

The dual bool makes it a little confusing, but the different bool checks serve different purposes.